src/datetime/serde.rs

```diff
@@ -1168,6 +1172,77 @@ pub mod ts_seconds_option {
     }
 }
 
+/// Ser/de to/from RFC 2822 strings
+///
+/// Intended for use with `serde`'s `with` attribute.
+///
+/// # Example:
+///
+/// ```rust
+/// use chrono::{DateTime, NaiveDate, FixedOffset, TimeZone};
+/// use serde_derive::{Deserialize, Serialize};
+///
+/// #[derive(Deserialize, Serialize)]
+/// struct Example {
+///     #[serde(with = "chrono::serde::rfc2822")]
+///     time: DateTime<FixedOffset>
+/// }
+/// let offset = 3600;
+/// let actual_time = NaiveDate::from_ymd_opt(2018, 5, 17).unwrap()
+///     .and_hms_opt(02, 04, 59).unwrap()
+///     .and_local_timezone(TimeZone::from_offset(&FixedOffset::east_opt(offset).unwrap())).unwrap();
+/// let to_serialize = Example {
+///     time: actual_time.clone(),
+/// };
+///
+/// let serialized = serde_json::to_string(&to_serialize).unwrap();
+/// assert_eq!(serialized, r#"{"time":"Thu, 17 May 2018 02:04:59 +0100"}"#);
+///
+/// let deserialized: Example = serde_json::from_str(&serialized).unwrap();
+/// assert_eq!(deserialized.time, actual_time);
+/// ```
+pub mod rfc2822 {
+    use crate::serde::Rfc2822Visitor;
+    use crate::{DateTime, FixedOffset};
+    use core::fmt;
+    use serde::{de, ser};
+
+    /// Serialize a datetime into an RFC 2822 formatted string, e.g. "01 Jun 2016 14:31:46 -0700"
+    ///
+    /// Intended for use with `serde`s `serialize_with` attribute.
+    pub fn serialize<S>(dt: &DateTime<FixedOffset>, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: ser::Serializer,
+    {
+        serializer.serialize_str(&dt.to_rfc2822())
```

Comment:
In `format/formatting.rs` there is a `write_rfc2822` function that works without allocating. Using that will make this serializer work on `no_std` targets like the others.

You will have to adjust its `#[cfg(feature = "alloc")]` to also enable it with the `serde` feature.